hera_mc/weather.py

```diff
@@ -0,0 +1,226 @@
+
+    return times_keep, vals_keep
+
+
+class WeatherData(MCDeclarativeBase):
+    """
+    Definition of weather table.
+
+    time: gps time of KAT weather sensor data, floored (BigInteger, part of primary_key).
+    variable: name of weather variable. One of the keys in weather_sensor_dict (String, part of primary_key)
+    value: measured value (Float)
+    """
+    __tablename__ = 'weather_data'
+    time = Column(BigInteger, primary_key=True)
+    variable = Column(String, nullable=False, primary_key=True)
+    value = Column(Float, nullable=False)
+
+    @classmethod
+    def create(cls, time, variable, value):
+        """
+        Create a new weather object.
+
+        Parameters:
+        ------------
+        time: astropy time object
+            astropy time object based on a timestamp from the katportal sensor.
+        variable: string
+            must be a key in weather_sensor_dict
+        value: float
+            value from the sensor associated with the variable
+        """
+        if not isinstance(time, Time):
+            raise ValueError('time must be an astropy Time object')
+        if variable not in weather_sensor_dict.keys():
+            raise ValueError('variable must be a key in weather_sensor_dict.')
+        if not isinstance(value, float):
+            raise ValueError('value must be a float.')
+        weather_time = floor(time.gps)
+
+        return cls(time=weather_time, variable=variable, value=value)
+
+
+@tornado.gen.coroutine
+def _helper_create_from_sensors(starttime, stoptime, variables=None):
+    """
+    Create a list of weather objects from sensor data using tornado server.
+
+    Parameters:
+    ------------
+    starttime: astropy time object
+        time to start getting history.
+    stoptime: astropy time object
+        time to stop getting history.
+    variable: string
+        variable to get history for. Must be a key in weather_sensor_dict,
+        defaults to all keys in weather_sensor_dict
+
+    Returns:
+    -----------
+    A list of WeatherData objects (only accessible via a yield call)
+    """
+    from katportalclient import KATPortalClient
+
+    if not isinstance(starttime, Time):
+        raise ValueError('starttime must be an astropy Time object')
+
+    if not isinstance(stoptime, Time):
+        raise ValueError('stoptime must be an astropy Time object')
+
+    if variables is None:
+        variables = weather_sensor_dict.keys()
+    elif not isinstance(variables, (list, tuple)):
+        variables = [variables]
+
+    sensor_names = []
+    sensor_var_dict = {}
+    for var in variables:
+        if var not in weather_sensor_dict.keys():
+            raise ValueError('variable must be a key in weather_sensor_dict')
+        sensor_names.append(weather_sensor_dict[var]['sensor_name'])
+        sensor_var_dict[weather_sensor_dict[var]['sensor_name']] = var
+
+    portal_client = KATPortalClient('http://portal.mkat.karoo.kat.ac.za/api/client',
```

Comment:
This is where the KAT URL is hard coded.  If we have to change it in the future, is this the only place it is?

Reply:
yes. Do you want me to put it at the top of the file?